### DISCUSSION TITLE
热更新问题请教一下🙏🙏🙏

### DISCUSSION THREAD

**Discussion by Atw-Lee** [umijs/umi#11880] [Q&A]
和umi一样。我使用turbo+dumi+father构建了一个monorepo。但是当我更新了ui包之后。dumi中的demo并没有热更新。我需要注释掉UI组件再放开才可以更新。我看到了monorepoRedirect这个属性，但是不知道如何使用，我该如何设置呢。这是我的目录结构：
.
├── LICENSE
├── README.md
├── apps
│   └── docs
│       ├── LICENSE
│       ├── README.md
│       ├── demo
│       │   └── bar
│       │       └── stereoscopic
│       │           └── index.tsx
│       ├── docs
│       │   ├── components
│       │   │   └── index.md
│       │   ├── guide
│       │   │   └── index.md
│       │   ├── index.en-US.md
│       │   └── index.md
│       ├── package.json
│       ├── public
│       │   └── logo.svg
│       └── tsconfig.json
├── commitlint.config.js
├── package.json
├── packages
│   ├── bar
│   │   ├── LICENSE
│   │   ├── es
│   │   │   ├── index.js
│   │   │   └── stereoscopic
│   │   │       └── index.js
│   │   ├── lib
│   │   │   ├── index.js
│   │   │   └── stereoscopic
│   │   │       └── index.js
│   │   ├── package.json
│   │   ├── src
│   │   │   ├── index.ts
│   │   │   └── stereoscopic
│   │   │       └── index.tsx
│   │   └── tsconfig.json
│   ├── core
│   │   ├── LICENSE
│   │   ├── es
│   │   │   ├── index.d.ts
│   │   │   ├── index.d.ts.map
│   │   │   ├── index.js
│   │   │   ├── type.d.ts
│   │   │   ├── type.d.ts.map
│   │   │   └── type.js
│   │   ├── lib
│   │   │   ├── index.d.ts
│   │   │   ├── index.d.ts.map
│   │   │   ├── index.js
│   │   │   ├── type.d.ts
│   │   │   ├── type.d.ts.map
│   │   │   └── type.js
│   │   ├── package.json
│   │   ├── src
│   │   │   ├── index.tsx
│   │   │   └── type.ts
│   │   └── tsconfig.json
│   └── tsconfig
│       ├── base.json
│       ├── package.json
│       └── react-library.json
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
└── turbo.json

**Comment by Atw-Lee** [comment]
是我的问题，monorepoRedirect方法可以。dumi中热更新了bar中的组件。但是如果bar又依赖于core组件，那我应该怎么办呢。

**Comment by Atw-Lee** [comment]
可以了，我在dumi中的package.json文件中增加了core的依赖。但是感觉有些奇怪的，我并不会用到core，只用到了bar。bar里面dependencies了core。是我想的有问题吗，可以讲解一下吗。

**Comment by fz6m** [comment]
现在 `monorepoRedirect` 这个功能只能重定向项目本体的直接依赖，直接依赖的二级和以后的依赖就没定位了，还没做这个功能，后面看看加一下吧。

**Comment by xuya227939** [comment]
@Atw-Lee  @fz6m 老哥，我也是跟你同样的架构，我在 demo 中引入 UI 组件，应该如何引入呢？现在引入 UI 组件，直接报错，Module not found: Error: Can't resolve 'tendata-ui' in '/home/jiang/桌面/workspace/tendata-ui/packages/button/examples'    ，没懂，还请指教。

**Reply by Atw-Lee** [reply]
https://github.com/Atw-Lee/prettier-charts     你瞅瞅吧，好久没弄了。

**Comment by xuya227939** [comment]
@Atw-Lee 多谢，已搞定
